app/src/main/java/com/concordia/mcga/utilities/pathfinding/SingleMapPathFinder.java

```diff
@@ -1,56 +1,61 @@
 package com.concordia.mcga.utilities.pathfinding;
 
-import org.json.JSONArray;
-
+import com.concordia.mcga.exceptions.MCGAPathFindingException;
+import com.concordia.mcga.utilities.pathfinding.PathFinderTile.Type;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.TreeSet;
+import org.json.JSONArray;
 
 /**
  * Pathfinding class, which runs the A* shortest pathfinding algorithm
  */
-public class PathFinder {
+public class SingleMapPathFinder {
 
     TiledMap map;
-    Set<PathFinderTile> openSet;
+    TreeSet<PathFinderTile> openSet;
```

Comment:
In this case, I would like to restrict the openSet to TreeSet, as it MUST be ordered (i.e. the code depends on it). Other cases I will change.